### PR TITLE
Add support pcl::PointXYZRGBA to pcl::VLPGrabber

### PR DESCRIPTION
**Pull request by UnaNancyOwen**
Add support <code>pcl::PointXYZRGBA</code> to <code>pcl::VLPGrabber</code>.
This change will be able to attach color to point cloud using <code>pcl::visualization::PointCloudColorHandlerRGBField</code>.

This pull request is a re-send #1738.
Please check the discussion about this pull request.
Thanks,

### CODE REVIEW

File: io/include/pcl/io/hdl_grabber.h [hunk 1]
```diff
@@ -183,12 +206,17 @@ namespace pcl
       float
       getMaximumDistanceThreshold ();
 
+      /** \brief Returns the maximum number of lasers
+      */
+      static uint8_t
+      getMaximumNumberOfLasers ();
```
Comment (taketwo) [inline io/include/pcl/io/hdl_grabber.h]: This method is shadowed in the deriving class. This is a recipe for troubles!
Suppose you have `HDLGrabber* grabber = new VLPGrabber;`.
What do you think `grabber->getMaximumNumberOfLasers()` will return?

Please change into virtual const.

Comment (SergioRAgostinho) [reply io/include/pcl/io/hdl_grabber.h]: > Suppose you have `HDLGrabber* grabber = new VLPGrabber;`

I argue that use case is not really applicable given the current inheritance scheme (poorly implemented).

Comment (taketwo) [reply io/include/pcl/io/hdl_grabber.h]: Well, I did not invent this, this is exactly use-case Tsukasa proposed: https://github.com/PointCloudLibrary/pcl/pull/1738#issuecomment-346638594

Comment (UnaNancyOwen) [reply io/include/pcl/io/hdl_grabber.h]: Does your proposal mean this?

```patch
- static uint8_t
- getMaximumNumberOfLasers ();
+ virtual uint8_t
+ getMaximumNumberOfLasers () const;

  uint8_t
- pcl::HDLGrabber::getMaximumNumberOfLasers ()
+ pcl::HDLGrabber::getMaximumNumberOfLasers () const
  {
    return (HDL_MAX_NUM_LASERS);
  }
```
```patch
- static uint8_t
- getMaximumNumberOfLasers ();
+ virtual uint8_t
+ getMaximumNumberOfLasers () const;

  uint8_t
- pcl::VLPGrabber::getMaximumNumberOfLasers ()
+ pcl::VLPGrabber::getMaximumNumberOfLasers () const
  {
    return (VLP_MAX_NUM_LASERS);
  }
```

Comment (taketwo) [reply io/include/pcl/io/hdl_grabber.h]: Yes

Comment (UnaNancyOwen) [reply io/include/pcl/io/hdl_grabber.h]: Fixed.

**Comment by SergioRAgostinho** [comment]
@taketwo Do you think it is worth to keep the `breaks ABI` label even when it is superseeded by the `breaks API` one? In this case you added it after. What was the rationale behind it?

**Comment by SergioRAgostinho** [comment]
Removing API breakage tag based on https://github.com/PointCloudLibrary/pcl/pull/2396#issuecomment-425278490 and https://github.com/PointCloudLibrary/pcl/pull/2396#issuecomment-425408719.

**Comment by SergioRAgostinho** [comment]
Can be considered API breakage because we renamed protected members. If people derive from the class their code will break. Namely `current_scan_xyzrgb_`, `current_sweep_xyzrgb_`, `scan_xyzrgb_signal_` and `sweep_xyzrgb_signal_`.

**Pull request merged**